TM1py/Objects/GitProject.py

```diff
@@ -1,42 +1,116 @@
+        if precondition:
+            raise NotImplemented("'precondition' property not implemented for TM1ProjectTask")
+
+        if not any([chore, process]):
+            raise ValueError("TM1ProjectTask must either have a 'Process' or a 'Chore' property")
+
+        if all([chore, process]):
+            raise ValueError("TM1ProjectTask must not have a 'Chore' and 'Process' property")
+
+        if all([chore, parameters]):
+            raise ValueError("TM1ProjectTask must not have a 'Chore' and 'Parameters' property")
+
+        self.task_name = task_name
+        self.chore = chore
+        self.process = process
+        self.parameters = parameters
+        self.dependencies = dependencies
+
+    def construct_body(self) -> Dict:
+        body = dict()
+
+        if self.dependencies:
+            body["Dependencies"] = self.dependencies
+
+        if self.chore:
+            body["Chore"] = self.chore
+
+        else:
+            body = {
+                "Process": f"Processes('{self.process}')",
+                "Parameters": self.parameters
+            }
+
+        return body
+
+    @classmethod
+    def from_dict(cls, task_name: str, task: Dict):
+        return cls(
+            task_name=task_name,
+            chore=task.get("Chore"),
+            process=task.get("Process"),
+            parameters=task.get("Parameters"),
+            dependencies=task.get("Dependencies"),
+            precondition=task.get("Precondition"),
+        )
+
+
 class TM1Project(TM1Object):
     """ Abstraction of Git tm1project
     """
 
-    def __init__(self,
-                 version: int = 1.0,
-                 name: Optional[str] = '',
-                 settings: Optional[dict] = None,
-                 tasks: Optional[dict] = None,
-                 objects: Optional[dict] = None,
-                 ignore: Optional[list] = None,
-                 files: Optional[list] = None,
-                 deployment: Optional[dict] = None,
-                 prepush: Optional[list] = None,
-                 postpush: Optional[list] = None,
-                 prepull: Optional[list] = None,
-                 postpull: Optional[list] = None,
-                 dependencies: Optional[list] = None,
-                 preconditions: Optional[str] = ''):
-        """_summary_
+    def __init__(
+            self,
+            version: int = 1.0,
+            name: Optional[str] = '',
+            settings: Optional[Dict] = None,
+            tasks: Optional[Dict[str, TM1ProjectTask]] = None,
+            objects: Optional[Dict] = None,
+            ignore: Optional[List] = None,
+            files: Optional[List] = None,
+            deployment: Optional[Dict] = None,
+            pre_push: Optional[List] = None,
+            post_push: Optional[List] = None,
+            pre_pull: Optional[List] = None,
+            post_pull: Optional[List] = None,
+            dependencies: Optional[List] = None,
+            preconditions: Optional[str] = ''):

```

Comment:
removed dependencies and preconditions as they are not a property of the Tm1project but to either objects and tasks